semgrep-core/cli/Main.ml

```diff
@@ -726,15 +733,17 @@ let json_of_exn e =
 (*****************************************************************************)
 
 (*s: function [[Main_semgrep_core.semgrep_with_rules]] *)
-let semgrep_with_rules rules files =
+let semgrep_with_rules ~with_opt_cache rules files =
```

Comment:
An alternative is to put this in Flag_semgrep.ml as a global, so you don't have to thread it all around.